src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/SelfHostDeployer.cs

```diff
@@ -97,9 +97,7 @@ protected async Task<(Uri url, CancellationToken hostExitToken)> StartSelfHostAs
                     var targetFramework = DeploymentParameters.TargetFramework ?? (DeploymentParameters.RuntimeFlavor == RuntimeFlavor.Clr ? "net461" : "netcoreapp2.0");
 
                     executableName = DotnetCommandName;
-                    // Set VerifyMatchingImplicitPackageVersion to disable errors when Microsoft.NETCore.App's version is overridden externally
-                    // This verification doesn't matter if we are skipping restore during tests.
-                    executableArgs = $"run --no-restore --framework {targetFramework} /p:VerifyMatchingImplicitPackageVersion=false {DotnetArgumentSeparator}";
+                    executableArgs = $"run --no-build --no-restore --framework {targetFramework} {DotnetArgumentSeparator}";
```

Comment:
So, is it safe to assume the application has already been built?

Reply:
That's the theory. I don't think even a local universe build would be enough to test it though, the CI runs tests differently.

Reply:
To test this, try doing this:
```
dotnet restore /p:MicrosoftNETCoreApp21PackageVersion=(something newer than what's in build/dependencies.props)
dotnet run --no-build
```
I want to know if the target that issues the VerifyMatchingImplicitPackageVersion error still runs if you pass `--no-build`.

Reply:
By the way, --no-build implies --no-restore